File: backend/src/main/java/com/example/spring_boot_ecommerce/entity/OrderItem.java
package com.example.spring_boot_ecommerce.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Column(name="unit_price")
    private BigDecimal unitPrice;

    @Column(name="quantity")
    private int quantity;

    @Column(name="product_id")
    private Long productId;

    // image
    @Column(name = "image_name")
    private String imageName;

    @Column(name = "image_type")
    private String imageType;

    @Column(name = "image_data", columnDefinition = "bytea")
    private byte[] imageData;  // remove @Lob

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
}

File: frontend/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageData: string;
    unitPrice: number;
    quantity: number;
    productId: number;

    constructor(cartItem: CartItem) {
        this.imageData = cartItem.imageData;
        this.unitPrice = cartItem.unitPrice;
        this.quantity = cartItem.quantity;
        this.productId = +cartItem.id;
    }


}